a10_octavia/controller/worker/flows/a10_listener_flows.py

```diff
@@ -92,19 +123,45 @@ def get_delete_listener_flow(self):
 
         return delete_listener_flow
 
+    def _get_delete_listener_subflow(self):
+        delete_listener_subflow = linear_flow.Flow(a10constants.DELETE_LISTENER_SUBFLOW)
+        delete_listener_subflow.add(lifecycle_tasks.ListenerToErrorOnRevertTask(
+            requires=constants.LISTENER))
+        delete_listener_subflow.add(a10_database_tasks.GetVThunderByLoadBalancer(
+            requires=constants.LOADBALANCER,
+            provides=a10constants.VTHUNDER))
+        delete_listener_subflow.add(*self.get_listener_delete_cert_template_subflow())
+        delete_listener_subflow.add(virtual_port_tasks.ListenerDelete(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER]))
+        return delete_listener_subflow
+
+    def _get_cert_delete_subflow(self):
+        get_delete_cert_subflow = linear_flow.Flow(a10constants.DELETE_CERT_TEMPLATE_SUBFLOW)
+        get_delete_cert_subflow.add(cert_tasks.SSLKeyDelete(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        get_delete_cert_subflow.add(cert_tasks.ClientSSLTemplateDelete(
+            requires=(a10constants.CERT_DATA, a10constants.VTHUNDER)))
+        return get_delete_cert_subflow
+
+    def get_listener_delete_cert_template_subflow(self):
+        get_listener_delete_cert_subflow = graph_flow.Flow(
+            a10constants.DELETE_LISTENER_CERT_SUBFLOW)
+        cert_delete_task = cert_tasks.SSLCertDelete(
+            requires=[constants.LOADBALANCER, constants.LISTENER, a10constants.VTHUNDER],
+            provides=a10constants.CERT_DATA)
+        cert_delete_template_task = self._get_cert_delete_subflow()
+        get_listener_delete_cert_subflow.add(cert_delete_task, cert_delete_template_task)
+        get_listener_delete_cert_subflow.link(
+            cert_delete_task, cert_delete_template_task, decider=self._is_terminated_https)
```

Comment:
This graph flow and decider should be in method _get_delete_listener_subflow. This is similar comment as in get_listener_create_cert_template_subflow.

Reply:
Please refer my above comments